### PR TITLE
Fix for rsp files on Windows.

### PR DESCRIPTION
**Pull request by cjatin**
File names starting with @ can not be escaped.

This is needed for really long command lines with tests on windows

### CODE REVIEW

File: bin/hipcc.pl [hunk 1]
```diff
@@ -371,6 +371,15 @@ BEGIN
         $hsacoVersion = $arg;
         $swallowArg = 1;
     }
+    if($arg =~ /\.rsp$/) {
```
Comment (yxsamliu) [inline bin/hipcc.pl:374]: Do you have an example response file that this patch is trying to fix? What is the error without this patch?

Comment (cjatin) [reply bin/hipcc.pl:374]: The error that we see is one hip-tests, GraphTests have grown in size and the final linkage line is too big on windows.

For now we have fixed it via a hack: https://github.com/ROCm-Developer-Tools/hip-tests/commit/fe9515a99a58ae38bcfb87b3a4f94123c3c21ff5 basically split it in two exe till we have a fix in place.

To fix this we need to enable rsp files inside hip-tests CMake (done via: https://github.com/ROCm-Developer-Tools/hip-tests/pull/360 ) but before that hipcc needs to escape rsp files and add proper encoding for windows.

Comment (cjatin) [reply bin/hipcc.pl:374]: Edit: we got same error for memory tests as well: https://github.com/ROCm-Developer-Tools/hip-tests/pull/362

**Review by kzhuravl — APPROVED** [review]
LGTM, but please run the psdb before merging.

**Pull request merged**